File: app/src/main/java/com/github/nein37/styleviewer/TextAppearanceAdapter.java
package com.github.nein37.styleviewer;


import android.content.res.TypedArray;
import android.databinding.DataBindingUtil;
import android.graphics.Color;
import android.support.v4.widget.TextViewCompat;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.github.nein37.styleviewer.databinding.ItemTextappearanceBinding;

import java.lang.reflect.Field;
import java.util.List;

public class TextAppearanceAdapter extends RecyclerView.Adapter<TextAppearanceAdapter.TextAppearanceViewHolder> {

    List<Field> textAppearanceList;

    public TextAppearanceAdapter(List<Field> textAppearanceList) {
        this.textAppearanceList = textAppearanceList;
    }

    @Override
    public TextAppearanceAdapter.TextAppearanceViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        ItemTextappearanceBinding bidning = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), R.layout.item_textappearance, parent, false);
        return new TextAppearanceViewHolder(bidning);
    }

    @Override
    public void onBindViewHolder(TextAppearanceAdapter.TextAppearanceViewHolder holder, int position) {
        Field field = textAppearanceList.get(position);
        holder.bind(field);
    }

    @Override
    public int getItemCount() {
        return textAppearanceList.size();
    }

    static class TextAppearanceViewHolder extends RecyclerView.ViewHolder {

        ItemTextappearanceBinding binding;

        public TextAppearanceViewHolder(ItemTextappearanceBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(Field field) {

            String styleName = field.getName().replace("_", ".");
            int styleId = 0;
            try {
                styleId = field.getInt(null);
            } catch (IllegalAccessException e) {
            }
            TextViewCompat.setTextAppearance(binding.styleName, styleId);
            binding.styleName.setText(styleName);

            int textColor = (0xFFFFFFFF & binding.styleName.getCurrentTextColor());
            binding.textColor.setText(binding.textColor.getResources().getString(R.string.text_color_format, textColor));
            if (textColor == Color.WHITE) {
                binding.getRoot().setBackgroundColor(Color.GRAY);
            } else {
                binding.getRoot().setBackgroundColor(Color.WHITE);
            }

            float textSize = binding.styleName.getTextSize();
            int textSizeSp = Math.round(textSize / binding.getRoot().getResources().getDisplayMetrics().scaledDensity);
            binding.textSize.setText(binding.textSize.getResources().getString(R.string.text_size_format, textSizeSp));

            boolean isBold = false;
            boolean isItalic = false;
            if (binding.styleName.getTypeface() != null) {
                isBold = binding.styleName.getTypeface().isBold();
                isItalic = binding.styleName.getTypeface().isItalic();
            }
            binding.isBold.setText(Boolean.toString(isBold));
            binding.isItalic.setText(Boolean.toString(isItalic));

            boolean isAllCaps = false;
            if (binding.styleName.getTransformationMethod() != null) {
                if (TextUtils.equals(binding.styleName.getTransformationMethod().getClass().getSimpleName(), "AllCapsTransformationMethod")) {
                    isAllCaps = true;
                }
            }
            binding.isAllCaps.setText(Boolean.toString(isAllCaps));
        }
    }
}

File: app/src/main/java/com/github/nein37/styleviewer/TextAppearanceAdapter.kt
package com.github.nein37.styleviewer


import android.content.res.TypedArray
import android.databinding.DataBindingUtil
import android.graphics.Color
import android.support.v4.widget.TextViewCompat
import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.ViewGroup

import com.github.nein37.styleviewer.databinding.ItemTextappearanceBinding
import com.github.nein37.styleviewer.repository.StyleEntity

import java.lang.reflect.Field

class TextAppearanceAdapter(internal var textAppearanceList: List<StyleEntity>) : RecyclerView.Adapter<TextAppearanceAdapter.TextAppearanceViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TextAppearanceAdapter.TextAppearanceViewHolder {
        val bidning = DataBindingUtil.inflate<ItemTextappearanceBinding>(LayoutInflater.from(parent.context), R.layout.item_textappearance, parent, false)
        return TextAppearanceViewHolder(bidning)
    }

    override fun onBindViewHolder(holder: TextAppearanceAdapter.TextAppearanceViewHolder, position: Int) {
        val entity = textAppearanceList[position]
        holder.bind(entity)
    }

    override fun getItemCount(): Int {
        return textAppearanceList.size
    }

    class TextAppearanceViewHolder(var binding: ItemTextappearanceBinding) : RecyclerView.ViewHolder(binding.root) {

        fun bind(entity: StyleEntity) {

            val styleName = entity.styleName.replace("_", ".")
            var styleId = entity.styleId;
            TextViewCompat.setTextAppearance(binding.styleName, styleId)
            binding.styleName.setText(styleName)

            val textColor = 0xFFFFFFFF.toInt() and binding.styleName.currentTextColor
            binding.textColor.text = binding.textColor.resources.getString(R.string.text_color_format, textColor)
            if (textColor == Color.WHITE) {
                binding.root.setBackgroundColor(Color.GRAY)
            } else {
                binding.root.setBackgroundColor(Color.WHITE)
            }

            val textSize = binding.styleName.textSize
            val textSizeSp = Math.round(textSize / binding.root.resources.displayMetrics.scaledDensity)
            binding.textSize.text = binding.textSize.resources.getString(R.string.text_size_format, textSizeSp)

            var isBold = false
            var isItalic = false
            if (binding.styleName.typeface != null) {
                isBold = binding.styleName.typeface.isBold
                isItalic = binding.styleName.typeface.isItalic
            }
            binding.isBold.text = java.lang.Boolean.toString(isBold)
            binding.isItalic.text = java.lang.Boolean.toString(isItalic)

            var isAllCaps = false
            if (binding.styleName.transformationMethod != null) {
                if (TextUtils.equals(binding.styleName.transformationMethod.javaClass.getSimpleName(), "AllCapsTransformationMethod")) {
                    isAllCaps = true
                }
            }
            binding.isAllCaps.text = java.lang.Boolean.toString(isAllCaps)
        }
    }
}
